OOP-mini-project/event.js

```diff
@@ -0,0 +1,94 @@
+// Part 1.2 - create event.js
+// Part 1.3 - event constructor
+class Event {
+    constructor(name, description) {
+      this.name = name;
+      this.description = description;
+      this.blurb = ""; // added blurb so as to not modify description
+      this.availableTickets = [];
+    }
+  
+    // Part 2.3 - create new ticket type w/ name and price
+    addAvailableTickets(typeName, typePrice) {
+      const typeObj1 = new TicketType(typeName, typePrice);
+      this.availableTickets.push(typeObj1);
+    }
+  
+    // Part 2.5 - print all tickets && Part 2.6 - display ticket types on HTML
+    allTickets() {
+      this.blurb += `<br><strong> - All tickets: </strong>`
+      for (let i = 0; i < this.availableTickets.length; i++) {
+        this.blurb += `<li>${i+1}. ${this.availableTickets[i].name} ($${this.availableTickets[i].price}) `
+      }
+      this.description += this.blurb;
+    }
+  
+    // Part 2.7 - search for tickets within a lower and upper bound
+    searchTickets(lower, upper) {
+      let count = 0; // counts matching tickets
+      this.blurb = `<br><strong> - Eligible Tickets: </strong>` ; // default
+      for (let i = 0; i < this.availableTickets.length; i++) { // iterates through availableTickets
+        if (this.availableTickets[i].price >= lower && this.availableTickets[i].price <= upper) { // if within range
+          count++;
+          this.blurb += `<li>${count}. ${this.availableTickets[i].name} ($${this.availableTickets[i].price}) `;
+        } 
```

Comment:
usually when you call array[index] more than once, set it to a meaningful variable so your code is less verbose and thus (usually) more readable